dashboard/src/utils/filterDataset.js

```diff
@@ -1,3 +1,12 @@
+/**
+ * Filter the List of Datasets based on Date Range or Dataset Name or both
```

Comment:
Looking at the body of the function, I'm not seeing how it could be one or the other -- it looks like the selection criteria is _both_ contains `searchKey` _and_ date within the specified bounds.  I'm not seeing any provision for having parameters omitted or otherwise unspecified.  So, you should probably tighten up this description.